### PR TITLE
feat: added accordion for sideNav

### LINKED ISSUE #10744: SideNav

### PR DESCRIPTION
**Pull request by ameeetgaikwad**
## Description
Updates the `SideNav` component to the new DS specifications

## Related Issue
Part of [Implementation of the new DS specifications](https://github.com/ethereum/ethereum-org-website/issues/9548)

fixes #10744

### CODE REVIEW

**Comment by pettinarip** [comment]
Thanks @ameeetgaikwad for the PR.

Marking this blocked until we merge #10936.

**Comment by wackerow** [comment]
#10936 Has updates pushed and ready for review... after can circle back to this

**Comment by minimalsm** [comment]
@ameeetgaikwad can you resolve conflicts here please?

@pettinarip @wackerow I think we can review this now?

**Comment by ameeetgaikwad** [comment]
> @ameeetgaikwad can you resolve conflicts here please?
> 
> @pettinarip @wackerow I think we can review this now?

sure!

**Comment by pettinarip** [comment]
Thanks @ameeetgaikwad, the build is failing 
```
8:15:44 AM: Failed to compile.
8:15:44 AM: ./src/components/SideNav.tsx
8:15:44 AM: 1:1  Error: Run autofix to sort these imports!  simple-import-sort/imports
```
Could you fix those issues? you can run `yarn lint:fix` to fix them.

===
Assigning the PR to myself for review.

**Comment by ameeetgaikwad** [comment]
@pettinarip 
image

**Review by corwintines — APPROVED** [review]
LGTM!

File: src/components/SideNav.tsx
Comment (pettinarip) [inline src/components/SideNav.tsx:170]: This does not change much from what we currently have. It just wraps the current logic into an `Accordion` & `AccordionItem` components that doesn't add anything new.

The idea of using the `Accordion` component was to see if we could replace and simplify the current custom logic we have (most of which lives in the `NavLink` component). And, at the same time, it could also inherit the styles already applied to meet the DS specifications. You can see the current styles [here](https://dev--63b7ea99632763723c7f4d6b.chromatic.com/?path=/story/molecules-disclosure-content-accordions--basic).

Some quick pseudo code
```tsx
<Accordion allowToggle>
  <AccordionItem>
    <h2>
      <AccordionButton>
        <Box as="span" flex="1" textAlign="left">
          Level 1 - 1
        </Box>
        <AccordionIcon />
      </AccordionButton>
    </h2>
    <AccordionPanel>
      <SideNavLink>Level 2 - link 1</SideNavLink>
      <SideNavLink>Level 2 - link 2</SideNavLink>
      <SideNavLink>Level 2 - link 3</SideNavLink>
    </AccordionPanel>
  </AccordionItem>
  <AccordionItem>
    <h2>
      <AccordionButton>
        <Box as="span" flex="1" textAlign="left">
          Level 1 - 2
        </Box>
        <AccordionIcon />
      </AccordionButton>
    </h2>
    <AccordionPanel>
      <SideNavLink>Level 2 - link 1</SideNavLink>
      <SideNavLink>Level 2 - link 2</SideNavLink>
      <SideNavLink>Level 2 - link 3</SideNavLink>

      <Accordion allowToggle>
        <AccordionItem>
          <h2>
            <AccordionButton>
              <Box as="span" flex="1" textAlign="left">
                Level 2 - nested accordion
              </Box>
              <AccordionIcon />
            </AccordionButton>
          </h2>
          <AccordionPanel>
            <SideNavLink>Level 2 - link 1</SideNavLink>
            <SideNavLink>Level 2 - link 2</SideNavLink>
            <SideNavLink>Level 2 - link 3</SideNavLink>
            
            .... // More nested accordions
          </AccordionPanel>
        </AccordionItem>
      </Accordion>
    </AccordionPanel>
  </AccordionItem>
</Accordion>
```

**Review by pettinarip — CHANGES_REQUESTED** [review]
Hey @ameeetgaikwad thanks for helping us out with this old PR again.

Unfortunately, the new changes don't add much to what we already had. I left an idea of what we could try to build using the `Accordion` component.

LMK if you need any help with this as it might be a bit challenging.

File: src/components/SideNav.tsx
Comment (ameeetgaikwad) [reply src/components/SideNav.tsx:170]: @pettinarip sure. I'll try to do it!

**Comment by pettinarip** [comment]
Closing in favor of #13063. @ameeetgaikwad thanks for the PR, feel free to share any thoughts on the new one.

**Pull request closed without merging**
